src/structures/idt.rs

```diff
@@ -779,34 +779,47 @@ impl<F> Entry<F> {
     }
 }
 
-macro_rules! impl_set_handler_fn {
-    ($h:ty) => {
+#[cfg(all(feature = "instructions", feature = "abi_x86_interrupt"))]
```

Comment:
I don't think we need this `cfg` for the `impl Entry<F>`, as its only the implementation of `HandlerFuncType` which has to be gated behind the feature.

Reply:
Specifically, this `impl<F: HandlerFuncType> Entry<F>` should only be gated behind `feature = "instructions"` and the `impl HandlerFuncType for $f` should be gated behind `feature = "abi_x86_interrupt"` (which is the gate we use to define the aliases for `HandlerFunc`, `HandlerFuncWithErrCode`, etc...)